ext/drbbdup/drbbdup.c

```diff
@@ -703,13 +757,112 @@ drbbdup_insert_dispatch(void *drcontext, instrlist_t *bb, instr_t *where,
     drbbdup_insert_landing_restoration(drcontext, bb, where, manager);
 }
 
+/* Returns whether or not additional cases should be handled poised by copy limit
+ * defined by the user.
+ */
+static bool
+drbbdup_do_dynamic_handling(drbbdup_manager_t *manager)
+{
+    drbbdup_case_t *drbbdup_case;
+    int i;
+    for (i = 0; i < opts.dup_limit; i++) {
+        drbbdup_case = &(manager->cases[i]);
+        /* Search for empty undefined slot. */
+        if (!drbbdup_case->is_defined)
+            return true;
+    }
+
+    return false;
+}
+
+/* Insert trigger for dynamic case handling. */
+static void
+drbbdup_insert_dynamic_handling(void *drcontext, app_pc translation_pc, void *tag,
+                                instrlist_t *bb, instr_t *where,
+                                drbbdup_manager_t *manager)
+{
+    instr_t *instr;
+    opnd_t opnd;
+
+    instr_t *done_label = INSTR_CREATE_label(drcontext);
+    opnd_t mask_opnd = opnd_create_reg(DRBBDUP_SCRATCH_REG);
+
+    /* Check whether case limit  has not been reached. */
+    if (drbbdup_do_dynamic_handling(manager)) {
+        drbbdup_case_t *default_info = &(manager->default_case);
```

Comment:
nit: Parenteses are not needed in many of these `&` cases throughout this patch: IMHO it is cleaner and easier to read w/o the parens, though that can be subjective.

Reply:
Done